### PR TITLE
Fold jingo_minify into our code.

### LINKED ISSUE #8532: Fold jingo_minify into addons-server
In preparation and as part of our upgrade to Django 1.11 we should fold jingo_minify into addons-server to eventually get rid of it and get more control over it.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This will simplify the upgrade to Django 1.11 and allow us to keep this
under control, upstream isn't properly maintained.

Fixes #8532

### CODE REVIEW

File: src/olympia/amo/management/commands/compress_assets.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+    def handle(self, **options):
+        self.build_id = self.generate_build_id(options.get('use_uuid', False))
+
+        if options.get('do_update_only', False):
+            self.update_hashes(update=True)
+            return
+
+        jar_path = (os.path.dirname(__file__), '..', '..', 'bin',
+                    'yuicompressor-2.4.7.jar')
```
Comment (diox) [inline src/olympia/amo/management/commands/compress_assets.py]: Ugh. I was happier not knowing.

File: src/olympia/amo/management/commands/compress_assets.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+                # Concat all the files.
+                tmp_concatted = '%s.tmp' % concatted_file
+                if len(files_all) == 0:
+                    raise CommandError(
+                        'No input files specified in '
+                        'MINIFY_BUNDLES["%s"]["%s"] in settings.py!' %
+                        (ftype, name)
+                    )
+                self._call(
+                    'cat %s > %s' % (' '.join(files_all), tmp_concatted),
```
Comment (diox) [inline src/olympia/amo/management/commands/compress_assets.py]: This I knew already and... well, this is terrible.

File: src/olympia/amo/management/commands/compress_assets.py (hunk 1 above)
Comment (EnTeQuAk) [reply src/olympia/amo/management/commands/compress_assets.py]: :rofl: I'm inclined to not change this for now though tbh, it's bad and needs a reasonable replacement sometime anyway.

File: src/olympia/amo/management/commands/compress_assets.py (hunk 2 above)
Comment (EnTeQuAk) [reply src/olympia/amo/management/commands/compress_assets.py]: :see_no_evil: it is indeed, crazily terrible. I didn't dare to touch any of that.

File: src/olympia/amo/management/commands/compress_assets.py (hunk 1 above)
Comment (EnTeQuAk) [reply src/olympia/amo/management/commands/compress_assets.py]: We're using uglifyjs though and this is properly configured.

File: src/olympia/amo/management/commands/compress_assets.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+        if options.get('do_update_only', False):
+            self.update_hashes(update=True)
+            return
+
+        jar_path = (os.path.dirname(__file__), '..', '..', 'bin',
+                    'yuicompressor-2.4.7.jar')
+        self.path_to_jar = os.path.realpath(os.path.join(*jar_path))
+
+        self.v = '-v' if options.get('verbosity', False) == '2' else ''
```
Comment (eviljeff) [inline src/olympia/amo/management/commands/compress_assets.py]: 🤣 What a variable name.  Especially as it's used as a boolean flag almost everywhere.  `self.verbose` and do `-v` when actually used?

File: src/olympia/amo/management/commands/compress_assets.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+
+        cachebust_imgs = getattr(settings, 'CACHEBUST_IMGS', False)
```
Comment (eviljeff) [inline src/olympia/amo/management/commands/compress_assets.py]: Shall we just drop this?  The setting is `True` everywhere.

File: src/olympia/amo/management/commands/compress_assets.py [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+                        concatted_file)
+                else:
+                    self.minify_skipped += 1
+
+        # Write out the hashes
+        self.update_hashes(options.get('add_timestamp', False))
+
+        if not self.v and self.minify_skipped:
+            print 'Unchanged files skipped for minification: %s' % (
+                self.minify_skipped)
```
Comment (eviljeff) [inline src/olympia/amo/management/commands/compress_assets.py]: can we wrap all `print xxx` as `print(xxx)` so it's automagically python3 compatible.

File: src/olympia/amo/management/commands/compress_assets.py [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+
+    def _get_url_or_path(self, item):
+        """
+        Determine whether this is a URL or a relative path.
+        """
+        if item.startswith('//'):
+            return 'http:%s' % item
+        elif item.startswith(('http', 'https')):
+            return item
+        return None
```
Comment (eviljeff) [inline src/olympia/amo/management/commands/compress_assets.py]: Ha, so if `item = 'foo/baa.css'` then `_get_url_or_path(item)` returns `None`.

File: src/olympia/lib/jingo_minify_helpers.py [hunk 7]
```diff
@@ -151,3 +149,45 @@ def build_ids(request):
     """A context processor for injecting the css/js build ids."""
     return {'BUILD_ID_CSS': BUILD_ID_CSS, 'BUILD_ID_JS': BUILD_ID_JS,
             'BUILD_ID_IMG': BUILD_ID_IMG}
+
+
+def get_media_root():
+    """Return STATIC_ROOT or MEDIA_ROOT depending on JINGO_MINIFY_USE_STATIC.
+    This allows projects using Django 1.4 to continue using the old
+    ways, but projects using Django 1.4 to use the new ways.
```
Comment (eviljeff) [inline src/olympia/lib/jingo_minify_helpers.py]: gotta keep that django1.4 compatibility!

File: src/olympia/amo/management/commands/compress_assets.py [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+            help='Add timestamp to hash')
+
+    def generate_build_id(self, use_uuid):
+        if use_uuid:
+            return uuid.uuid4().hex[:8]
+        else:
+            root = getattr(settings, 'JINGO_MINIFY_ASSETS_GIT_ROOT', '.')
+            git_bin = getattr(settings, 'GIT_BIN', 'git')
+            return check_output(
+                [git_bin, '-C', root, 'rev-parse', '--short', 'HEAD']).strip()
```
Comment (eviljeff) [inline src/olympia/amo/management/commands/compress_assets.py]: both places we call this command we do `compress_assets --use-uuid` so may as well drop the git support and simplify this code.

File: src/olympia/amo/management/commands/compress_assets.py (hunk 3 above)
Comment (EnTeQuAk) [reply src/olympia/amo/management/commands/compress_assets.py]: done, in the new PR I just opened. Don't ask me why, I should have re-opened this one.

File: src/olympia/amo/management/commands/compress_assets.py (hunk 4 above)
Comment (EnTeQuAk) [reply src/olympia/amo/management/commands/compress_assets.py]: Also done in the new PR

**Pull request closed without merging**